src/utils/createValidator.ts

```diff
@@ -19,6 +24,8 @@ export function createValidator<V, D, E>(data?: D, name?: string | string[]) {
             name: Array.isArray(name) ? name.join('.') : name
           })
         };
+      } else if (isPromiseLike(checkResult)) {
```

Comment:
It would be nice to be able to add test cases for the changes here.

Reply:
Good point added a test.